x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curation/curation_logic.test.ts

```diff
@@ -136,6 +148,104 @@ describe('CurationLogic', () => {
         });
       });
     });
+
+    describe('setPromotedIds', () => {
+      it('should set promotedIds state & promotedDocumentsLoading to true', () => {
+        mount();
+
+        CurationLogic.actions.setPromotedIds(['hello', 'world']);
+
+        expect(CurationLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          promotedIds: ['hello', 'world'],
+          promotedDocumentsLoading: true,
+        });
+      });
+    });
+
+    describe('addPromotedId', () => {
+      it('should set promotedIds state & promotedDocumentsLoading to true', () => {
```

Comment:
I'm fairly certain you'll hate this, but thought it might be a cool way to dry up these tests.


```ts
              
              const itShouldChangePromotedIdsState = (from, to, callback) => {
                it('should set promotedIds state & promotedDocumentsLoading to true', () => {
                   mount({ promotedIds: from });

                   callback();

                   expect(CurationLogic.values).toEqual({
                     ...DEFAULT_VALUES,
                     promotedIds: to,
                     promotedDocumentsLoading: true,
                   });
                 });
               }

               describe('addPromotedId', () => {
                 itShouldChangePromotedIdsState(['hello'], ['hello', 'world'], () => {
                   CurationLogic.actions.addPromotedId('world');
                 });
               }

               describe('setPromotedIds', () => {
                 itShouldChangePromotedIdsState([], ['hello', 'world'], () => {
                   CurationLogic.actions.addPromotedId(['hello', 'world']);
                 });
               }

               describe('removePromotedId', () => {
                 itShouldChangePromotedIdsState(['hello', 'deleteme', 'world'], ['hello', 'world'], () => {
                   CurationLogic.actions.removePromotedId('deleteme');
                 });
               }
```

Or some variant of that
```ts
             it('should set promotedIds state & promotedDocumentsLoading to true', () => {
                 expectAction(() => {
                   CurationLogic.actions.removePromotedId('deleteme');
                 }).toChangePromotedIds(
                   ['hello', 'deleteme', 'world'],
                   ['hello', 'world']
                 );
               }
```

Just riffing now, what if we had a generic "expectAction" we could use in specs:
```ts
             it('should set promotedIds state & promotedDocumentsLoading to true', () => {
                 expectAction(() => {
                   CurationLogic.actions.removePromotedId('deleteme');
                 }).toChangeState(
                   {
                     promotedIds: ['hello', 'deleteme', 'world'],
                     promotedDocumentsLoading: false
                   },
                   {
                     promotedIds: ['hello', 'world'],
                     promotedDocumentsLoading: true
                   }
                 );
               }
```

It would handle mounting with defaults, and then running expectations on values.

Reply:
Actually, I should clarify - this isn't a change request for the this PR right?

Reply:
Nope